engine/gameobject/src/gameobject/res_gameobject.cpp

```diff
@@ -110,7 +110,8 @@ namespace dmGameObject
                                                                               type,
                                                                               type_index,
                                                                               component_desc.m_Position,
-                                                                              component_desc.m_Rotation);
+                                                                              component_desc.m_Rotation,
+                                                                              component_desc.m_Scale);
```

Comment:
It seems like we could do `component_desc.m_HasScale ? component_desc.m_Scale : Vector3(1.0f, 1.0f, 1.0f)` here and skip writing the value to the binary in when building the project? If I understand correctly, this would be the only place we'd need to change in the runtime for that to work.